python/lsst/daf/butler/registry/queries/_query.py

```diff
@@ -322,6 +324,20 @@ def explain_no_results(
                 f"{self._filtered_by_join} result rows were filtered out because "
                 "one or more regions did not overlap."
             )
+        if not followup or self._filtered_by_join or self._filtered_by_where:
```

Comment:
Personally I prefer if (not followup) or self._filtered_by_join or self._filtered_by_where: simply because it makes the meaning immediately clear for future readers who may not remember the order such statements are evaluated (i.e. to not get confused with not (x or y or z). Up to you though since the code works properly as-is.